### PR TITLE
Move to single ESM file, and add workflow for test

### PR DESCRIPTION
**Pull request by SanderElias**
- Move to a single ESM file, as separating those doesn't add any value
- keep separated CJS, as this one does not load the bulk if not needed
- Update readme with advice on how to use in ESM mode
- add Github Action workflow to run test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +16,16 @@ The polyfill will only be loaded if the URLPattern doesn't already exist on the
 // CJS module loading (Node.js)
 require("urlpattern-polyfill");
 
-// ESM module loading (Node.js)
+// ESM module loading (Node.js and browser)
```
Comment (kenchris) [inline README.md]: // Conditional ESM module loading (Node.js and web browser)

File: README.md [hunk 2]
```diff
@@ -16,11 +16,16 @@ The polyfill will only be loaded if the URLPattern doesn't already exist on the
...
+if (!globalThis.URLPattern) {
+  await import("urlpattern-polyfill");
+}
+/** 
+ * The above is the recommend way to load the ESM module.
```
Comment (kenchris) [inline README.md]: * The above is the recommended way to load the ESM module, as it only 
 * loads it on demand, thus when not natively supported by the runtime or 
 * already polyfilled.

File: README.md [hunk 3]
```diff
@@ -16,11 +16,16 @@ The polyfill will only be loaded if the URLPattern doesn't already exist on the
...
 require("urlpattern-polyfill");
 
-// ESM module loading (Node.js)
+// ESM module loading (Node.js and browser)
+if (!globalThis.URLPattern) {
+  await import("urlpattern-polyfill");
+}
+/** 
+ * The above is the recommend way to load the ESM module. 
+ * However, when you can't use top level await,
```
Comment (kenchris) [inline README.md]: or you always want to load it and replace any existing URLPattern object, you can ...

Comment (SanderElias) [reply README.md]: The condition is still in the code, and will not overwrite a preexisting implementation.
I will add en export to the url-pattern, so when someone would want to do this, it becomes possible

Comment (kenchris) [reply README.md]: Yeah that is correct, but we should do that and document it

**Review by kenchris — APPROVED** [review]
Can you please bump the major version

**Review by kenchris — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (SanderElias) [reply README.md]: exported url-pattern, and updated readme

**Pull request merged**